x-pack/plugins/fleet/server/routes/agent/upgrade_handler.ts

```diff
@@ -146,3 +168,67 @@ const checkSourceUriAllowed = (sourceUri?: string) => {
     );
   }
 };
+
+// Check the installed fleet server versions
+// Allow upgrading if the agents to upgrade include fleet server agents
+const checkFleetServerVersion = async (
```

Comment:
I am wondering if we should move this check inside the upgrade service in the `isUpgradable` function